floodsub.go

```diff
@@ -171,7 +175,19 @@ func (p *PubSub) processLoop(ctx context.Context) {
 				continue
 			}
 		case msg := <-p.publish:
-			p.maybePublishMessage(p.host.ID(), msg.Message)
+			subs := p.getSubscriptions(msg) // call before goroutine!
+			go func() {
+				if p.validate(subs, msg) {
```

Comment:
Shouldn't at the very least be some debugging output when we fail validation?

Reply:
I don't know - is that log-worthy? To me it seems that a validator returning false is not very exceptional - and if it is, the caller can still log from inside the validator.

Reply:
I think it is, it may indicate a bug or an attack and these things are nice to always nice to know.

Reply:
I guess it depends on what the validator is used for. Anyway, added logging [here](https://github.com/libp2p/go-floodsub/pull/45/commits/af6cff63be0db01c0672a8041c1d4975c10eebe5#diff-c37f865b99d25cd944a588b8b2e90cbeR347)